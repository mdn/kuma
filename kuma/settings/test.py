from .common import *  # noqa

ATTACHMENT_HOST = 'mdn-local.mozillademos.org'

DEBUG = False
CELERY_ALWAYS_EAGER = True
CELERY_EAGER_PROPAGATES_EXCEPTIONS = True
ES_LIVE_INDEX = False
ES_DISABLED = True

PASSWORD_HASHERS = (
    'django.contrib.auth.hashers.SHA1PasswordHasher',
)

INSTALLED_APPS += (
    'kuma.core.tests.taggit_extras',
    'kuma.actioncounters.tests',
)

DEMO_UPLOADS_ROOT = '/home/vagrant/uploads/demos'
