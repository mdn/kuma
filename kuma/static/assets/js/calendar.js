$(document).ready(function() {

    var markers = [];
    var $mapCanvas = $('#map_canvas');
    var address;

    $('#upcoming tbody tr').each(function(){
        var $self = $(this);
        var html = $self.children('td.date').html() + '<br/>' + $self.children('td.conference').html() + '<br/>';

        // Add the new marker
        markers.push({
            address: $self.children('td.location').text(),
            icon: {
                image: '/static/assets/img/icn-tool-mozilla.png',
                iconsize: [32, 26],
                iconanchor: [16, 13],
                infowindowanchor: [24, 0]
            },
            html: html
        });
    });

    // Start rendering the map with a default position.
    $mapCanvas.gMap({ longitude: 1, latitude: 10, markers: markers, zoom: 2 });

    if ('geolocation' in navigator) {
        navigator.geolocation.getCurrentPosition(function (position) {
            // If we're able and permitted to access position, move the map
            // center and zoom in a bit.
            var coords = position.coords;
            $mapCanvas.data('gmap').setCenter(new GLatLng(coords.latitude, coords.longitude), 4);
        });
    }

});
