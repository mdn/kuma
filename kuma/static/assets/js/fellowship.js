(function (win, $) {
    'use strict';

        var $cta = $('.cta:not(a)');
        var $cta_links = $cta.find('a');
        // allow focus
        $cta.attr('tabindex', 0).addClass('js');
        // when activated add the class to show and remove listeners which preventDefault
        function activate(caller) {
            var $caller = $(caller);
            var $calling_cta;
            if($caller.hasClass('cta')){
                $calling_cta = $caller;
            } else {
                $calling_cta = $caller.parents('.cta');
            }
            $calling_cta.addClass('active').off();
            $calling_cta.find('a').off();
        }
        // listen for tap
        $cta.on('focus', function() {
            activate(this);
        });
        // listen for focus
        $cta_links.on('touchstart', function(event) {
            event.preventDefault();
            activate(this);
        });

})(window, jQuery);
