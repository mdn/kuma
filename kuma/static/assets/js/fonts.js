(function(win, doc, $) {
    /*
        Font Loading
    */

    if(win.waffle && win.waffle.flag_is_active('web_font_loader')) {

        // what fonts are we loading?
        var fonts = [
            {
                "name" : "Open Sans Light",
                "class" : "ffo-opensanslight",
                "varient" : [
                    {"weight" : "normal"},
                    {"weight" : "bold"},
                    {"style" : "italic"}
                ]
            },
            {
                "name" : "Open Sans",
                "class" : "ffo-opensans",
                "varient" : [
                    {"weight" : "normal"},
                    {"weight" : "bold"},
                    {"style" : "italic"}
                ]
            }
        ];

        // to collect the results of the observers as they return
        var fontsLoaded = {};

        // timout for all observers
        var ffoTimeout = 3000;

        // starts observers for all fonts in fonts array
        function ffoLoad() {
            // for each font
            $.each(fonts, function() {
                // add the class to the fontsLoaded object
                var fontClass = this.class;
                fontsLoaded[fontClass] = {};

                // get name for later use
                var fontName = this.name;

                // for each varient of this font
                $.each(this.varient, function(index) {
                    var fontVarient = this;
                    var fontVarientNumber = index;
                    // add an object to the fontsLoaded object
                    fontsLoaded[fontClass][fontVarientNumber] = false;

                    // add observer for this varient
                    var ffObserver = new FontFaceObserver(fontName, fontVarient);

                    // add callbacks for this observer
                    ffObserver.check(null, ffoTimeout).then(function () {
                        // update fontsLoaded object to show it loaded
                        fontsLoaded[fontClass][fontVarientNumber] = true;
                        ffoCheck();
                    }, function () {
                        // timeout
                        ffoCheck();
                    });

                });
            });
        }

        function ffoCheck(){
            var allLoaded = true;

            // loop through fontsLoaded looking to see if they've loaded
            $.each(fontsLoaded, function(){
                var isLoaded = true;
                $.each(this, function(){
                    if(this == false) {
                        isLoaded = false;
                    }
                })
                if(isLoaded == false) {
                    allLoaded = false;
                }
            });

            // if all fonts are loaded swap font in
            if(allLoaded) {
                ffoFinished();

                // set cookie so fonts are displayed automatically with the server's help
                var expires = new Date();
                expires.setDate(expires.getDate() + 1);
                document.cookie = 'ffo=true; expires=' + expires + '; path=/';
            }
        }

        function ffoFinished() {
            ffoSwap();
            tabzillaLoad();
        }

        function ffoSwap() {
            // swaps in available fonts
            $.each(fontsLoaded, function(index) {
                var fontAttribute = index;
                var fontLoaded = true;
                $.each(this, function() {
                    if(this == false) {
                        fontLoaded = false;
                    }
                })
                if(fontLoaded == true) {
                    $('html').attr("data-"+fontAttribute, true);
                }
            });
        }

        function tabzillaLoad() {
            $tabzilla = $('#tabzilla');

            if(!$tabzilla.length) return;

            $('<link />').attr({
                href: '//mozorg.cdn.mozilla.net/media/css/tabzilla-min.css',
                type: 'text/css',
                rel: 'stylesheet'
            }).on('load', function() {

                $('#tabzilla').addClass('loaded');

                $.ajax({
                    url: '//mozorg.cdn.mozilla.net/en-US/tabzilla/tabzilla.js',
                    dataType: 'script',
                    cache: true
                });
            }).prependTo(doc.head);
        };

        // load the observer plug in
        $.getScript('/static/assets/compressed/js/fontfaceobserver.js')
        .done(ffoLoad)
        .fail(ffoFinished);

    };

})(window, document, jQuery);
