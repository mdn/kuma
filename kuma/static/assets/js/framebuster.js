// Quick and dirty way to bust out of a frame / iframe.
if (top.location != self.location) {
    top.location = self.location.href;
}
