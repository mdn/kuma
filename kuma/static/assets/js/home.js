(function($) {
    'use strict';

    // Create the demos slider
    $('.home-demos-list').css('height', 'auto').owlCarousel({ lazyLoad: true });

    // Track search submissions with Optimizely
    $('#home-search-form').on('submit', function() {
        mdn.optimizely.push(['trackEvent', 'submit-homepage-search-form']);
    });

})(jQuery);
