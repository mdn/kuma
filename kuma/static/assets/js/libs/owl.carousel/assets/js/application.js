$(function(){
    window.prettyPrint && prettyPrint();
}(window.jQuery));

jQuery(document).ready(function($) {

	//Download Link Highlight
	if($("body").data("page")==="frontpage"){
		$(window).scroll(function(){
			var scrolled = $(window).scrollTop();
		    var downloadLink = $("#top-nav").find(".download")
		    if(scrolled >= 420){
		    	downloadLink.addClass("download-on");
		    } else if (scrolled < 420){
		    	downloadLink.removeClass("download-on");
		    }
		})
	}

	$('#myTab a').click(function (e) {
	  e.preventDefault();
	  $(this).tab('show');
	})


});
