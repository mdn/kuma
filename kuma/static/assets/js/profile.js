//
// Profile view and edit enhancements
//
(function () {
    'use strict';

    var DEBOUNCE_DELAY = 25;

    // Translate multiple rapid calls to a function into a single call after a
    // short delay. Also serves to allow UI updates to complete due to yielding
    // the event loop.
    //
    // This seems to paper over some odd timing bugs where checkboxes aren't
    // detected, and certain events aren't caught.
    function debounce(func, wait, immediate) {
        var timeout;
        return function() {
            var context = this, args = arguments;
            var later = function() {
                timeout = null;
                if (!immediate) func.apply(context, args);
            };
            var callNow = immediate && !timeout;
            clearTimeout(timeout);
            timeout = setTimeout(later, wait);
            if (callNow) func.apply(context, args);
        };
    }

    // Rebuild the list of expertise tags and checkboxes.
    var rebuildExpertiseTaglist = debounce(function () {
        var taglist = $('#tags-expertise');
        var interests = $('#id_profile-interests');
        var i_tags = interests.val().split(",");

        // Completely rebuild the list of expertise tags. Seems wasteful, but
        // the number of elements should be relatively tiny vs the code to do
        // it more surgically.
        taglist.empty();
        $.each(i_tags, function (idx, tag) {
            tag = $.trim(tag);
            if(window.INTEREST_SUGGESTIONS.indexOf(tag) === -1) return;

            taglist.append('<li class="tag-expert">' +
                '<label for="expert-' + idx + '">' +
                '<input type="checkbox" name="expert-' + idx + '" ' +
                    'id="expert-'+idx+'" value="' + tag + '"> ' + tag +
                '</label></li>');
        });

        // Do this mutual update, so any checkboxes that have disappeared
        // also get removed from the field.
        updateTaglistFromField();
        updateFieldFromTaglist();
    });

    // Update the checked tags in expertise tag list from the text field
    var updateTaglistFromField = debounce(function () {
        var taglist = $('#tags-expertise');
        var expertise = $('#id_profile-expertise');
        var eTags = expertise.val().split(',');

        $('#tags-expertise .tag-expert input[type=checkbox]').removeAttr('checked');
        $.each(eTags, function(idx, tag) {
            tag = $.trim(tag);
            $('#tags-expertise .tag-expert input[value=' + tag + ']').attr('checked', 'checked');
        });
    });

    // Update the expertise text field from checked boxes in tag list
    var updateFieldFromTaglist = debounce(function () {
        var tags = $('#tags-expertise .tag-expert input[type=checkbox]:checked')
            .map(function () { return $(this).val(); })
            .get().join(',');
        $('#id_profile-expertise').val(tags);
    });

    $(document).ready(function(){

        // Convert interests text field into a tag-it widget
        $('#id_profile-interests').hide()
            .after('<ul id="tagit-interests"></ul>')
            .change(rebuildExpertiseTaglist);

        $('#tagit-interests').tagit({
            availableTags: window.INTEREST_SUGGESTIONS,
            singleField: true,
            singleFieldNode: $('#id_profile-interests'),
            onTagAdded: rebuildExpertiseTaglist,
            onTagRemoved: rebuildExpertiseTaglist,
            onTagClicked: rebuildExpertiseTaglist
        });

        // Set the new tag element text
        $('#tagit-interests .tagit-new input').attr('placeholder', gettext('New interest...'));

        // Convert the expertise text field into tag list with checkboxes sync'd to
        // interests
        $("#id_profile-expertise").hide().after("<ul id='tags-expertise' class='tags'></ul>");

        $('#tags-expertise').click(updateFieldFromTaglist);
        rebuildExpertiseTaglist();

        // word count
        $('.wordcount').each(function(i, el){

            var $el = $(el);
            var placeholder = $el.find('.counter');
            var limit = parseInt(placeholder.text(), 10);
            var currcount = 0;
            var field = $el.children('textarea');

            function updateWordCount() {
                var words = $.trim(field.val()).split(' ');
                var color = placeholder.parent().css('color');
                var invalidColor = '#900';
                var length;

                if(words[0] === ''){ words.length = 0; }
                currcount = limit - words.length;
                placeholder.text(currcount);

                length = words.length;

                if(length >= limit && color !== invalidColor) {
                    placeholder.parent().css('color', invalidColor);
                }
                else if(words.length < limit && color === invalidColor) {
                    placeholder.parent().css('color', '');
                }
            }

            updateWordCount();
            field.keypress(updateWordCount);
        });

        // Update "Other Profiles", preventing "blank" submissions
        $('#profiles input').mozPlaceholder();
    });
})();
