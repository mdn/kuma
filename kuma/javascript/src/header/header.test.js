//@flow
import React from 'react';
import { create } from 'react-test-renderer';

import { fakeDocumentData } from '../document.test.js';
import Header from './header.jsx';

test('Header snapshot', () => {
    const header = create(<Header document={fakeDocumentData} />);
    expect(header.toJSON()).toMatchSnapshot();
});
