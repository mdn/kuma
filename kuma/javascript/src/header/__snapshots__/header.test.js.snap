// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
.emotion-2 {
  grid-area: S;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2px 8px;
  border: solid 2px #333;
  border-radius: 20px;
  background-color: #fafafa;
  box-sizing: border-box;
  height: 40px;
  min-width: 80px;
  -webkit-flex: 2 1;
  -ms-flex: 2 1;
  flex: 2 1;
}

.emotion-0 {
  fill: #333;
  -webkit-flex: 0 0 20px;
  -ms-flex: 0 0 20px;
  flex: 0 0 20px;
}

.emotion-1 {
  border-width: 0 !important;
  font-size: 14px;
  -webkit-flex: 1 1 100px;
  -ms-flex: 1 1 100px;
  flex: 1 1 100px;
  min-width: 60px;
  max-width: 320px;
  color: #000;
  background-color: #fafafa !important;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-self: center;
  font-size: 1em;
  font-weight: bold;
  color: black;
  margin-left: 24px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #eee;
}

.emotion-3:focus {
  outline-offset: -3px;
}

<header
  className="page-header"
>
  <a
    aria-label="MDN Web Docs"
    className="logo"
    href="/en-US/"
  >
    <svg mock />
  </a>
  <nav
    className="main-nav"
    role="navigation"
  >
    <ul>
      <li
        className="top-level-entry-container"
      >
        <button
          aria-haspopup="true"
          className="top-level-entry"
          onTouchStart={[Function]}
          type="button"
        >
          Technologies
          <span
            aria-hidden="true"
            className="main-menu-arrow"
          >
            ‚ñº
          </span>
        </button>
        <ul
          className={null}
        >
          <li
            data-item="Technologies"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web"
            >
              Technologies Overview
            </a>
          </li>
          <li
            data-item="Technologies"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web/HTML"
            >
              HTML
            </a>
          </li>
          <li
            data-item="Technologies"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web/CSS"
            >
              CSS
            </a>
          </li>
          <li
            data-item="Technologies"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web/JavaScript"
            >
              JavaScript
            </a>
          </li>
          <li
            data-item="Technologies"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web/Guide/Graphics"
            >
              Graphics
            </a>
          </li>
          <li
            data-item="Technologies"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web/HTTP"
            >
              HTTP
            </a>
          </li>
          <li
            data-item="Technologies"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web/API"
            >
              APIs / DOM
            </a>
          </li>
          <li
            data-item="Technologies"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Mozilla/Add-ons/WebExtensions"
            >
              Browser Extensions
            </a>
          </li>
          <li
            data-item="Technologies"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web/MathML"
            >
              MathML
            </a>
          </li>
        </ul>
      </li>
      <li
        className="top-level-entry-container"
      >
        <button
          aria-haspopup="true"
          className="top-level-entry"
          onTouchStart={[Function]}
          type="button"
        >
          References & Guides
          <span
            aria-hidden="true"
            className="main-menu-arrow"
          >
            ‚ñº
          </span>
        </button>
        <ul
          className={null}
        >
          <li
            data-item="References & Guides"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Learn"
            >
              Learn web development
            </a>
          </li>
          <li
            data-item="References & Guides"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web/Tutorials"
            >
              Tutorials
            </a>
          </li>
          <li
            data-item="References & Guides"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web/Reference"
            >
              References
            </a>
          </li>
          <li
            data-item="References & Guides"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web/Guide"
            >
              Developer Guides
            </a>
          </li>
          <li
            data-item="References & Guides"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web/Accessibility"
            >
              Accessibility
            </a>
          </li>
          <li
            data-item="References & Guides"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Games"
            >
              Game development
            </a>
          </li>
          <li
            data-item="References & Guides"
            role="menuitem"
          >
            <a
              href="/en-US/docs/Web"
            >
              ...more docs
            </a>
          </li>
        </ul>
      </li>
      <li
        className="top-level-entry-container"
      >
        <button
          aria-haspopup="true"
          className="top-level-entry"
          onTouchStart={[Function]}
          type="button"
        >
          Feedback
          <span
            aria-hidden="true"
            className="main-menu-arrow"
          >
            ‚ñº
          </span>
        </button>
        <ul
          className={null}
        >
          <li
            data-item="Feedback"
            role="menuitem"
          >
            <a
              href="/en-US/docs/MDN/Feedback"
            >
              Send Feedback
            </a>
          </li>
          <li
            data-item="Feedback"
            role="menuitem"
          >
            <a
              href="https://support.mozilla.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Get Firefox help
               üåê
            </a>
          </li>
          <li
            data-item="Feedback"
            role="menuitem"
          >
            <a
              href="https://stackoverflow.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Get web development help
               üåê
            </a>
          </li>
          <li
            data-item="Feedback"
            role="menuitem"
          >
            <a
              href="/en-US/docs/MDN/Community"
            >
              Join the MDN community
            </a>
          </li>
          <li
            data-item="Feedback"
            role="menuitem"
          >
            <a
              href="https://github.com/mdn/sprints/issues/new?template=issue-template.md&projects=mdn/sprints/2&labels=user-report&title=%2Fen-US%2Fdocs%2Ftest"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report a content problem
               üåê
            </a>
          </li>
          <li
            data-item="Feedback"
            role="menuitem"
          >
            <a
              href="https://bugzilla.mozilla.org/form.mdn"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report a bug
               üåê
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </nav>
  <form
    action="/en-US/search"
    className="header-search emotion-2"
    id="nav-main-search"
    method="get"
    role="search"
  >
    <svg mock
      className="emotion-0"
    />
    <label
      className="visually-hidden"
      htmlFor="main-q"
    >
      Search MDN
    </label>
    <input
      className="emotion-1"
      id="main-q"
      name="q"
      placeholder="Search MDN"
      type="search"
    />
  </form>
  <a
    className="emotion-3"
    data-service="GitHub"
    href="/users/github/login/?next=/[fake%20absolute%20url]"
    rel="nofollow"
  >
    Sign in
  </a>
</header>
`;
